classes/search/cmsfield.php

```diff
@@ -253,4 +158,60 @@ protected function get_data($id) {
         }
         return $this->cmsdata[$id];
     }
+
+    /**
+     * Returns true if this area uses file indexing.
+     *
+     * @return bool
+     */
+    public function uses_file_indexing() {
+        return true;
+    }
+
+    /**
+     * Return the context info required to index files for
+     * this search area.
+     *
+     * @return array
+     */
+    public function get_search_fileareas() {
+        return ['value'];
+    }
+
+    /**
+     * Add the cms file attachments.
+     *
+     * @param document $document The current document
+     * @return null
+     */
+    public function attach_files($document) {
+        global $DB;
+
+        $fileareas = $this->get_search_fileareas();
+        // File is in "customfield_file" for component, "value" for filearea, and for customfield data id for itemid.
+        $contextid = \context_system::instance()->id;
```

Comment:
is it correct context ID for files?

Reply:
For file customfiled, it's using context which is from custom field handler context.
`$context = $this->get_field()->get_handler()->get_configuration_context();`
https://github.com/andrewhancox/moodle-customfield_file/blob/11a770aa46fa170d498a062a6cf3c75def145bc4/classes/data_controller.php#L76

The `get_configuration_context` function in mod_cms is using `\context_system::instance();`, so the context for the file should be correct.
https://github.com/catalyst/moodle-mod_cms/blob/ba37892fd6da36d80d9d7b76114540af02bd2e65/classes/customfield/cmsfield_handler.php#L41

Reply:
I added the issue https://github.com/catalyst/moodle-mod_cms/issues/136